src/shared/__tests__/libs/iota/transfers.spec.js

```diff
@@ -43,30 +43,127 @@ describe('libs: iota/transfers', () => {
         });
 
         describe('value transactions', () => {
-            it('should return transfer value', () => {
-                expect(
-                    getTransferValue(
-                        [
-                            { value: -10, currentIndex: 0, lastIndex: 4 },
-                            { value: -1, currentIndex: 1, lastIndex: 4 },
-                            { value: -40, currentIndex: 2, lastIndex: 4 },
-                        ],
-                        [{ value: 12, currentIndex: 3, lastIndex: 4 }, { value: 39, currentIndex: 4, lastIndex: 4 }],
-                    ),
-                ).to.equal(12);
+            describe('with any input address belong to user addresses', () => {
+                it('should return a difference of inputs and remainder', () => {
+                    expect(
+                        getTransferValue(
+                            [
+                                {
+                                    value: -10,
+                                    currentIndex: 0,
+                                    lastIndex: 4,
+                                    address: 'A'.repeat(81),
+                                },
+                                {
+                                    value: -1,
+                                    currentIndex: 1,
+                                    lastIndex: 4,
+                                    address: 'B'.repeat(81),
+                                },
+                                {
+                                    value: -40,
+                                    currentIndex: 2,
+                                    lastIndex: 4,
+                                    address: 'C'.repeat(81),
+                                },
+                            ],
+                            [
+                                {
+                                    value: 12,
+                                    currentIndex: 3,
+                                    lastIndex: 4,
+                                    address: 'Z'.repeat(81),
+                                },
+                                {
+                                    value: 39,
+                                    currentIndex: 4,
+                                    lastIndex: 4,
+                                    address: 'D'.repeat(81),
+                                },
+                            ],
+                            ownAddresses,
+                        ),
+                    ).to.equal(12);
+                });
+            });
+
+            describe('with no input addresses belong to user addresses', () => {
+                it('should return a sum of all user output addresses', () => {
+                    expect(
+                        getTransferValue(
+                            [
+                                {
+                                    value: -10,
+                                    currentIndex: 0,
```

Comment:
Wouldn't the transaction with a `currentIndex` of 0 be an output/have a positive value?

Reply:
I don't think it's necessary but I'll sort the indexes. Good catch.

Reply:
Yeah, I didn't think it would affect anything really

Reply:
Actually this isn't a big deal, we can merge this anyway